modules/user_accounts/jsx/userAccountsIndex.js

```diff
@@ -0,0 +1,174 @@
+          return row;
+        });
+        this.setState({data});
+      })
+      .catch((error) => {
+        this.setState({error: true});
+        console.error(error);
+      });
+  }
+
+  /**
+   * Modify behaviour of specified column cells in the Data Table component
+   *
+   * @param {string} column - column name
+   * @param {string} cell - cell content
+   * @param {object} row - row content indexed by column
+   *
+   * @return {*} a formated table cell for a given column
+   */
+  formatColumn(column, cell, row) {
+    let url;
+    let result = <td>{cell}</td>;
+    switch (column) {
+      case 'Site':
+        // If user has multiple sites, join array of sites into string
+        result = <td>{cell.join('; ')}</td>;
+        break;
+      case 'Username':
+        url = loris.BaseURL + '/user_accounts/edit_user/' + row.Username;
+        result = <td><a href ={url}>{cell}</a></td>;
+        break;
+      case 'Active':
+        if (row.Active === 'Y') {
+          result = <td>Yes</td>;
+        } else if (row.Active === 'N') {
+          result = <td>No</td>;
+        }
+        break;
+      case 'Pending Approval':
+        if (row['Pending Approval'] === 'Y') {
+          result = <td>Yes</td>;
+        } else if (row['Pending Approval'] === 'N') {
+          result = <td>No</td>;
+        }
+        break;
+    }
+    return result;
+  }
+
+  addUser() {
+    location.href='/user_accounts/edit_user/';
+  }
+
+  render() {
+    // If error occurs, return a message.
+    // XXX: Replace this with a UI component for 500 errors.
+    if (this.state.error) {
+      return <h3>An error occured while loading the page.</h3>;
+    }
+
+    // Waiting for async data to load
+    if (!this.state.isLoaded) {
+      return <Loader/>;
+    }
+
+   /**
+    * XXX: Currently, the order of these fields MUST match the order of the
+    * queried columns in _setupVariables() in userAccounts.class.inc
+    */
+    const options = this.state.data.fieldOptions;
+    const fields = [
+      {label: 'Site', show: true, filter: {
+        name: 'site',
+        type: 'select',
+        options: options.sites,
+      }},
+      {label: 'Username', show: true, filter: {
+        name: 'username',
+        type: 'text',
+      }},
+      {label: 'Full Name', show: true, filter: {
+        name: 'fullName',
+        type: 'text',
+      }},
+      {label: 'Email', show: true, filter: {
+        name: 'email',
+        type: 'text',
+      }},
+      {label: 'Active', show: true, filter: {
+        name: 'active',
+        type: 'select',
+        options: options.actives,
+      }},
+      {label: 'Pending Approval', show: true, filter: {
+        name: 'pendingApproval',
+        type: 'select',
+        options: options.pendingApprovals,
+      }},
+    ];
+    const actions = [
+      {name: 'addUser', label: 'Add User', action: this.addUser},
+    ];
+
+    return (
+      <div id="userAccountsIndex">
+        <div id="userAccountsFilter">
```

Comment:
What's the purpose of these `id`s?

Reply:
for description of the divs

Reply:
Okay, but how are they being used? `FilterableDataTable` already receives a name prop which is used to set an `id` on the filter. This index file is already a `UserAccountsIndex` component, so why is there a need for a description? Are these `id`s used as targets for testing? We should only put `id`s that are required by the needs of our current application.

Reply:
when inspecting elements, I find it useful to have ids that literally identify which divs are for what. in this case, that's what the ids are: to name the divs. they don't provide any function per say but I don't see why that's a problem.  the initial thinking was to use the ids as selectors in testing, but now that's no longer necessary. however, in the future, we might need to target those divs for some reason, and then the ids are already there to select

<img width="364" alt="screenshot 2019-01-14 13 12 41" src="https://user-images.githubusercontent.com/17415878/51131857-d2f64e00-17fe-11e9-937f-ad194b37210e.png">

what reason is there for not allowing ids unless they are required for targets?

Reply:
I'm not sure that ids are designed to name the divs. According to W3Schools they are used to specifically target elements for use by css stylesheets, javascript, etc. for manipulation. The resources I've looked into recommend that if they are not necessary, don't use them. If we aren't limiting what elements should have `id`s to those that necessitate them, then `id`s could be placed everywhere. If they are necessary in the future, they can be added then! 

https://css-tricks.com/the-difference-between-id-and-class/
https://stackoverflow.com/questions/42384788/should-we-always-specify-an-id-attribute-in-html
https://stackoverflow.com/questions/16153986/adding-id-attribute-to-all-html-elements-of-a-web-application